### DISCUSSION TITLE
Firefox for Android is going to support extensions by the end of 2023!

### DISCUSSION THREAD

**Discussion by BroJac5246** [ScratchAddons/ScratchAddons#6512] [Discussion]
Announced today:
https://blog.mozilla.org/addons/2023/08/10/prepare-your-firefox-desktop-extension-for-the-upcoming-android-release/

Learn more:
https://www.androidpolice.com/firefox-bringing-back-full-extension-support-android/

Scratch Addons officially supports Firefox, so bringing the extension to Android presumably wouldn't be too hard. However, the extension is not optimized for touchscreens, as brought up in #5160.

This is a pretty big deal, but the question is: will we support it? Since we really need to improve on touchscreen support, this could be good motivation, and I can't see why we wouldn't want to support this. If we do add support, this will be a game-changer for Scratch on mobile devices, particularly Android tablets.

**Comment by DNin01** [comment]
Support for mobile devices would be much more worth-it if Chrome also supported extensions on mobile, which last time I heard it did not.

**Reply by BroJac5246** [reply]
Mobile versions of Chrome do not support extensions. That's why this is so exciting, it's the first major browser to support extensions on Android (and obviously Apple won't allow extensions on iOS/iPadOS, except for maybe when the EU's new app store rules go into place).

**Reply by WorldLanguages** [reply]
Chrome extensions on mobile probably won't be a thing because of adblock 🤔

**Comment by WorldLanguages** [comment]
Can we create an issue about this?

**Reply by BroJac5246** [reply]
#6789 (just so it's linked)

**Comment by BroJac5246** [comment]
Oh also we should keep in mind that at some point in the future someone with probably release an iOS browser in the EU that supports extensions

**Reply by WorldLanguages** [reply]
The situation is quite complex, I wouldn't be so sure. It's possible regulators may need to determine if Apple is actually compliant or not.

**Reply by DNin01** [reply]
Why not just support iOS/iPadOS via a Safari extension?

**Reply by BroJac5246** [reply]
> Why not just support iOS/iPadOS via a Safari extension?

A) Because then we have to support Safari, which of course that takes effort.
B) I'm not saying we shouldn't not support Safari (particularly considering Macs), just that we should keep this in mind. It seems unlikely that the US or anywhere else will pass a DMA equivalent, but if a Blink- or Gecko-powered browser is released (which seems likely) then the extension may be automatically compatible.

**Reply by BroJac5246** [reply]
> The situation is quite complex, I wouldn't be so sure. It's possible regulators may need to determine if Apple is actually compliant or not.

Regardless, it is all but certain that non-WebKit browsers will be allowed, given that Apple had already announced it.

**Reply by BroJac5246** [reply]
Edit: I'm pretty sure the DMA doesn't affect iPadOS, just iOS

**Reply by GrahamSH-LLK** [reply]
That doesn't make sense to me
Edit: yeah it absolutely applies to ipads.

**Reply by BroJac5246** [reply]
> That doesn't make sense to me
> Edit: yeah it absolutely applies to ipads.

https://apple.stackexchange.com/questions/469358/will-the-eu-dma-changes-spare-ipad-and-macos-devices

**Reply by BroJac5246** [reply]
It will now apply to iPads within 6 months. https://9to5mac.com/2024/04/29/eu-ipad-digital-markets-act/
